addons/binding/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/handler/GardenaAccountHandler.java

```diff
@@ -178,9 +178,9 @@ public void onDeviceUpdated(Device device) {
                 gardenaThingHandler.updateSettings(device);
                 gardenaThingHandler.updateStatus(device);
             } catch (GardenaException ex) {
-                logger.error("There is something wrong with your thing, please recreate the thing {}",
-                        gardenaThing.getUID(), ex);
-                updateStatus(ThingStatus.OFFLINE);
+                logger.error("There is something wrong with your thing '{}', please check or recreate it: {}",
+                        gardenaThing.getUID(), ex.getMessage());
```

Comment:
I don't know what is catched and how valuable the underlying stacktrace is but if its for example `NullPointerException`s they tend to miss a message, so you will be printing the message `null` which will make it troublesome to debug, in general I think its better to log the complete stacktrace, maybe on another level if desired?

Reply:
No, these will never be NPEs, because we are in the catch for `GardenaException`s :-)

Reply:
I was referring to the cause of the `GardenaException` so the stack trace of the cause of the `GardenaException` 

So this line ;-)

https://github.com/openhab/openhab2-addons/blob/a483e5759c168b4cc800f39f98a595ab26e8fc1e/addons/binding/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartImpl.java#L422

In that case it is not likely that there is an NPE hiding there but we do lose any other context such as exact type of the exception and the underlying stacktrace. I fear that if we catch/log it like this we log nothing in a case such as a `GardenaUnauthorizedException` because it has no message and we discard the stacktrace and the type. 

So in general I don't like this form of logging :-)

Up to you what you want to do in this case

Reply:
I agree that the overall error handling is not optimal here - also the message to "recreate a thing" is not really ok; the implementation should instead handle such situations gracefully and continue to work as good as possible. Only in completely unexpected situations, the exception should be logged as an error, so that it can be reported as a bug - but asking the user to fix his setup is weird (this should result in an CONFIGURATION_ERROR as I did it here).